api/spec/spec_handler.go

```diff
@@ -398,6 +399,14 @@ func (d *specHandler) UpdateSpecFromOpts(opts map[string]string, spec *api.Volum
 					spec.Xattr = api.Xattr_COW_ON_DEMAND
 				}
 			}
+		case api.SpecDirectIo:
+			if directIo, err := strconv.ParseBool(v); err != nil {
+				return nil, nil, nil, err
+			} else {
+				if directIo {
+					spec.Xattr |= api.Xattr_DIRECT_IO
```

Comment:
Are xattr flags? It is not commented in that manner. You may want to use hex value and comment the volumespec accordingly. Also, it is too bad it is already committed, but it would be good to instead have xattr be something like `xttar_options` where it was an array of options instead of C-style flags. Makes it easier to do in  gRPC from any language to use an array of options instead of flags.